dts/arm/nxp/nxp_imx95_m7.dtsi

```diff
@@ -94,6 +94,7 @@
 			prescaler = <16>;
 			status = "disabled";
 			#pwm-cells = <3>;
+			always-on-clock;
```

Comment:
Why not simply have the clock controller handling the `IMX95_CLK_BUSWAKEUP` clock treat `clock_control_on()` calls gracefully?

Reply:
The clock controller is abstracted away by SCMI and thus doesn't have a notion of `IMX95_CLK_BUSWAKEUP`
https://github.com/zephyrproject-rtos/zephyr/blob/7fb92c9eb125f722da96a9df56622bd0b745f9eb/drivers/firmware/scmi/clk.c

I suppose for a board specific solution something could work would be to define a `fixed-clock` with a predefined frequency so that the `kinetis-tpm` driver wouldn't fail on initialization. 

The approach in this PR kept `clock_control_get_rate` API intact thus avoiding having predefined frequencies but also at the cost adding the dts entry/complexity.